src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Sve.cs

```diff
@@ -3386,5 +3673,110 @@ public new abstract class Arm64 : AdvSimd.Arm64
         ///   ZIP1 Presult.D, Pop1.D, Pop2.D
         /// </summary>
         public static unsafe Vector<ulong> ZipLow(Vector<ulong> left, Vector<ulong> right) => ZipLow(left, right);
+
+
+
+        private static int AddSaturateScalar(int left, int right)
+        {
+            int result = (int)(left + right);
+            bool overflow = (right < 0) ? (result > left) : (result < right);
+            return overflow ? (result > 0 ? int.MinValue : int.MaxValue) : result;
+        }
+
+        private static long AddSaturateScalar(long left, long right)
+        {
+            long result = (long)(left + right);
+            bool overflow = (right < 0) ? (result > left) : (result < right);
+            return overflow ? (result > 0 ? long.MinValue : long.MaxValue) : result;
+        }
+
+        private static uint AddSaturateScalar(uint left, uint right)
+        {
+            uint result = (uint)(left + right);
+            bool overflow = (result < left);
+            return overflow ? uint.MaxValue : result;
+        }
+
+        private static ulong AddSaturateScalar(ulong left, ulong right)
+        {
+            ulong result = (ulong)(left + right);
+            bool overflow = (result < left);
+            return overflow ? ulong.MaxValue : result;
+        }
+
+        private static int SubtractSaturateScalar(int left, int right)
+        {
+            int result = (int)(left - right);
+            bool overflow = (right < 0) ? (result < left) : (result > left);
+            return overflow ? (result > 0 ? int.MinValue : int.MaxValue) : result;
+        }
+
+        private static long SubtractSaturateScalar(long left, long right)
+        {
+            long result = (long)(left - right);
+            bool overflow = (right < 0) ? (result < left) : (result > left);
+            return overflow ? (result > 0 ? long.MinValue : long.MaxValue) : result;
+        }
+
+        private static uint SubtractSaturateScalar(uint left, uint right)
+        {
+            uint result = (uint)(left - right);
+            bool overflow = (left < right);
```

Comment:
this can be `left < result`, but by doing `left < right`, we can return early:

```c#
if (left < right) return uint.MinValue;
return (left - right);
```